### ISSUE TITLE
row-types treated as strings by cl-postgres

### ISSUE THREAD

**Issue by slyrus** [marijnh/Postmodern#86]
Given the following setup:

```
;; fill in as appropriate
(defparameter *db-settings*
  `(:host "HOST"
    :username "USER"
    :password "PASSWORD"
    :database-name "DATABASE"))

(defmacro with-db-connection ((conn-var) &body body)
  (let ((host-var (gensym))
        (username-var (gensym))
        (password-var (gensym))
        (database-name-var (gensym)))
    `(destructuring-bind (&key
                          ((:host ,host-var))
                          ((:username ,username-var))
                          ((:password ,password-var))
                          ((:database-name ,database-name-var)))
      *db-settings*
      (let ((,conn-var (cl-postgres:open-database
                        ,database-name-var ,username-var ,password-var ,host-var)))
        (unwind-protect
             (progn ,@body)
          (when ,conn-var (cl-postgres:close-database ,conn-var)))))))
```
the following query works fine:

```
(with-db-connection (conn)
  (cl-postgres:exec-query
   conn
   "select 22::integer, 44.5::double precision, 'abcde'::varchar, true::boolean, 4.5::numeric(5,2)"
   'cl-postgres:list-row-reader))
```
but this one has an unexpected result:

```
(with-db-connection (conn)
  (cl-postgres:exec-query
   conn
   "select (1, 2);"
   'cl-postgres:list-row-reader))
```
This returns the following string:
(("(1,2)"))

I'd like to get something like ((1,2)) back instead. Is this the fault of the list-row-reader or in something lower down in, e.g., the protocol for talking to the postgres database?

**Comment by slyrus** [comment]
hmm... sorry about the lousy formatting above.

**Comment by marijnh** [comment]
Types for which no reader exists are returned as strings. This is by design (i.e. it's a lot better than crashing, and for some types can still allow people to do useful things with the values)

**Comment by slyrus** [comment]
Well, yeah, but the point is to add readers for those types so that the user doesn't have to do a bunch of parsing work before doing those useful things. You seem to be implying that my version is likely to crash. I don't believe that to be the cases.

**Comment by marijnh** [comment]
No, I'm saying that returning strings is intentional. I'm not sure what 'your version' is supposed to refer to. If it's #87, sure, adding support for more types is a good thing.

**Comment by slyrus** [comment]
Yes, this "issue" is addressed by Pull Request #87.
